### PR TITLE
Improve S2699 (`assertions-in-tests`): Support Sinon.JS assertion statements

### LINKED ISSUE #3265: Improve S2699 (`assertions-in-tests`): Support Sinon.JS assertion statements
Reported in https://community.sonarsource.com/t/sonarqube-not-recognizing-sinon-add-at-least-one-assertion-to-this-test-case/68269

### PR DESCRIPTION
**Pull request by ilia-kebets-sonarsource**
Fixes #3265

### CODE REVIEW

**Review by yassin-kammoun-sonarsource — COMMENTED** [review]
Could you update the RSPEC and mention Sinon.js as well? Currently, it says the following:

> This rule raises an issue when the assertion library chai is imported but no assertion is used in a test.

**Comment by ilia-kebets-sonarsource** [comment]
the RSPEC PR: https://github.com/SonarSource/rspec/pull/2961

File: packages/jsts/src/rules/helpers/sinon.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import { Rule } from 'eslint';
+import * as estree from 'estree';
+import { getFullyQualifiedName, getImportDeclarations, getRequireCalls } from '.';
+
+export namespace Sinon {
+  export function isImported(context: Rule.RuleContext): boolean {
+    return (
+      getRequireCalls(context).some(
+        r => r.arguments[0].type === 'Literal' && r.arguments[0].value === 'chai',
+      ) || getImportDeclarations(context).some(i => i.source.value === 'chai')
```
Comment (yassin-kammoun-sonarsource) [inline packages/jsts/src/rules/helpers/sinon.ts:29]: `'chai'`!

Comment (yassin-kammoun-sonarsource) [reply packages/jsts/src/rules/helpers/sinon.ts:29]: Probably the reason why the coverage is a bit low.

Comment (ilia-kebets-sonarsource) [reply packages/jsts/src/rules/helpers/sinon.ts]: my god... Thanks!

**Comment by ilia-kebets-sonarsource** [comment]
I will replace the separate file with something more explicit like `sinon.fixture.js` and `chai.fixture.js` when the fixture is ready.

**Review by yassin-kammoun-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**